### PR TITLE
Fix pretty-printing of by-name and repeated parameter types.

### PR DESCRIPTION
**Pull request by olafurpg**

### CODE REVIEW

File: scalameta/semanticdb-scalac/src/test/scala/scala/meta/tests/SemanticSuite.scala [hunk 1]
```diff
@@ -3,6 +3,18 @@ package scala.meta.tests
 import scala.meta._
 import scala.meta.internal.semanticdb.SemanticdbMode
 
+// Contributing tips:
```
Comment (olafurpg) [inline scalameta/semanticdb-scalac/src/test/scala/scala/meta/tests/SemanticSuite.scala]: @fikovnik Anything else you can recommend for future contributors? Or any other question you'd like to see addressed here? :)

Comment (fikovnik) [reply scalameta/semanticdb-scalac/src/test/scala/scala/meta/tests/SemanticSuite.scala]: You guys actually did really nice set up. I used did exactly that - started with a simple copy of `SemanticSuite` (added a `println` of the obtained variable in `synthetics`) and ran sbt with `-jvmdebug` with attached debuggr. It was very smooth!

Sorry, I did not realized the alphabetic naming of the test objects :-|

File: scalameta/semanticdb-scalac/src/test/scala/scala/meta/tests/SemanticSuite.scala [hunk 2]
```diff
@@ -3,6 +3,18 @@ package scala.meta.tests
...
+//   add YYY.scala to your .gitignore, and run `> ~semanticdbScala/testOnly *YYY`.
+//   That should give you a tight edit/run/debug cycle.
+// - Go to DatabaseSuite and uncomment the "println" right above
+//   "assertDenotationSignaturesAreParseable", that will print out the obtained
+//   output into the console. You can copy-paste that output to put into the
+//   expected part of the  multiline strings.
+// - Try to follow the alphabetical order of the enclosing object, at the time
+//   of this writing the latest object is `object ad`, so the next object should
+//   be `object ae`.
+// - glhf, and if you have any questions don't hestiate to ask in the gitter channel :)
```
Comment (xeno-by) [inline scalameta/semanticdb-scalac/src/test/scala/scala/meta/tests/SemanticSuite.scala]: `hesitate`.

File: scalameta/semanticdb-scalac/src/test/scala/scala/meta/tests/SemanticSuite.scala (hunk 1 above)
Comment (xeno-by) [inline scalameta/semanticdb-scalac/src/test/scala/scala/meta/tests/SemanticSuite.scala]: Great idea! 👍

**Pull request merged**